bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/handler/JablotronJa100FHandler.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.jablotron.internal.handler;
+
+import static org.openhab.binding.jablotron.JablotronBindingConstants.CHANNEL_LAST_CHECK_TIME;
+
+import java.util.List;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.jablotron.internal.model.JablotronHistoryDataEvent;
+import org.openhab.binding.jablotron.internal.model.ja100f.JablotronGetPGResponse;
+import org.openhab.binding.jablotron.internal.model.ja100f.JablotronGetSectionsResponse;
+import org.openhab.binding.jablotron.internal.model.ja100f.JablotronSection;
+import org.openhab.binding.jablotron.internal.model.ja100f.JablotronState;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JablotronJa100FHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class JablotronJa100FHandler extends JablotronAlarmHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JablotronJa100FHandler.class);
+
+    public JablotronJa100FHandler(Thing thing, String alarmName) {
+        super(thing, alarmName);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().startsWith("SEC-") && command instanceof StringType) {
+            if ("PARTIAL_ARM".equals(command.toString())) {
+                controlComponent(channelUID.getId(), "CONTROL-SECTION", "DISARM");
+            }
+            scheduler.execute(() -> controlComponent(channelUID.getId(), "CONTROL-SECTION", command.toString()));
+        }
+
+        if (channelUID.getId().startsWith("PG-") && command instanceof OnOffType) {
+            scheduler.execute(() -> controlComponent(channelUID.getId(), "CONTROL-PG", command.toString()));
+        }
+    }
```

Comment:
their API is very slow, often experiences timeouts. Since I can update the status only of the whole alarm including all channels, calling this update method for each channel would kill this binding.

Reply:
This challenges many bindings. For this use case `ExpiringCache` has been introduced. See https://www.openhab.org/javadoc/v2.5/org/eclipse/smarthome/core/cache/expiringcache

Reply:
ok, implemented the expiring cache